### PR TITLE
Port liabilities

### PR DESCRIPTION
**Pull request by tatyree**
This is a port of the liabilities workflow from the prototype and includes payment and updating.

### CODE REVIEW

File: app/controllers/liabilities_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class LiabilitiesController < ApplicationController
+  def pay
+    operation = CreatePayment.new(params[:id])
+
+    target = unless operation.error?
```
Comment (digitalronin) [inline app/controllers/liabilities_controller.rb]: Elsewhere in the code, the pattern seems to be;
```
    if some_error?
      response = error_reponse
    else
      response = its_all_good
    end
````
So, for consistency and readability, I think it we could use the same pattern here, i.e.;
````
    if operation.error?
      target = root_path
      flash[:alert] = t('.govpay_api_error')
    else
      target = operation.payment_url
    end
````

File: app/models/fee_liability.rb [hunk 2]
```diff
@@ -21,6 +21,10 @@ def paid?
     govpay_payment_status == 'success'
   end
 
+  def failed?
```
Comment (digitalronin) [inline app/models/fee_liability.rb]: To me, "failed? == true" implies that we tried to do something, and we were unsuccessful. As written, are we saying that all fee liabilities start in the failed state, and stay that way until they are paid? If so, maybe we should consider changing our terminology. What do you think?

File: app/controllers/liabilities_controller.rb (hunk 1 above)
Comment (tatyree) [inline app/controllers/liabilities_controller.rb]: Already done, just haven't pushed it yet. Rubocop picked me up on that one when the code coverage passed.

File: app/models/fee_liability.rb (hunk 2 above)
Comment (tatyree) [inline app/models/fee_liability.rb]: Seems sensible.  I added it as a convenience method to be used when checking the status in #post_pay.

Comment (digitalronin) [inline app/models/fee_liability.rb]: Would something like this work?

````
  def failed?
    status_known? && !paid?
  end
````
...or is that being too picky?

Comment (tatyree) [inline app/models/fee_liability.rb]: Perfect.  Much clearer.

**Pull request merged**